File: server/src/main/java/de/scameronde/chat/businesstypes/ChatMessageLog.java
package de.scameronde.chat.businesstypes;

public class ChatMessageLog {
  String chatMessageLog;

  public ChatMessageLog() {

  }

  public ChatMessageLog(String messageLog) {
    this.chatMessageLog = messageLog;
  }

  public String getChatMessageLog() {
    return chatMessageLog;
  }

  public void setChatMessageLog(String messageLog) {
    this.chatMessageLog = messageLog;
  }
}

File: client/client/Businesstypes/ChatMessageLog.hs
{-# LANGUAGE AllowAmbiguousTypes        #-}
{-# LANGUAGE DataKinds                  #-}
{-# LANGUAGE DeriveGeneric              #-}
{-# LANGUAGE ScopedTypeVariables        #-}
{-# LANGUAGE TypeFamilies               #-}
{-# LANGUAGE TypeOperators              #-}
module Businesstypes.ChatMessageLog where

import           Data.Aeson
import           GHC.Generics
import           Miso.String             hiding ( toLower
                                                , toUpper
                                                , stripPrefix
                                                )


newtype ChatMessageLog = ChatMessageLog
  { messageLog :: MisoString
  } deriving (Show, Eq, Generic)
instance ToJSON ChatMessageLog
instance FromJSON ChatMessageLog
